src/main/java/org/pac4j/vertx/verticle/DemoServerVerticle.java

```diff
@@ -121,7 +121,13 @@ public void start() {
         // Requires authentication endpoint without specific authenticator attached
         addProtectedEndpointWithoutAuthorizer("/protected/index.html", "", router);
 
-        router.get("/index.html").handler(DemoHandlers.indexHandler(config));
+        // Direct basic auth authentication (web services)
+        addProtectedEndpointWithoutAuthorizer("/dba/index.html", "DirectBasicAuthClient", router);
```

Comment:
Session saving is tricky, because I’ve allowed the vert.x config to manage whether or not we wish to save user info in the session and this is either global (all vert.x auth endpoints) or not. This is one of the areas where vertx-web technically conflicts with pac4j. If we try to make pac4j overrule vert.x on this one we will break issue #55 on pac4j again (breaking an expected vert.x behaviour). The point is, for vert.x auth using the standard UserSessionHandler (which most will adopt), either you store the current context user in the session for any authentication, or you do so for no authentication (by not using the UserSessionManager art all). It would be possible to provide an overridden UserSessionHandler which might resolve this, by also storing a “store user in session” flag on the context which can be set by the VertxProfileManager. However, that means we’re providing an alternative to an existing vert.x-web class which could create more of a maintenance headache (e.g. any bugs fixed in that class don’t automatically come through to us)

Cheers,

Jez


> On 29 Dec 2015, at 13:20, LELEU Jérôme <notifications@github.com> wrote:
> 
> In src/main/java/org/pac4j/vertx/verticle/DemoServerVerticle.java <https://github.com/pac4j/vertx-pac4j-demo/pull/35#discussion_r48539623>:
> 
> > @@ -121,7 +121,13 @@ public void start() {
> >          // Requires authentication endpoint without specific authenticator attached
> >          addProtectedEndpointWithoutAuthorizer("/protected/index.html", "", router);
> >  
> > -        router.get("/index.html").handler(DemoHandlers.indexHandler(config));
> > +        // Direct basic auth authentication (web services)
> > +        addProtectedEndpointWithoutAuthorizer("/dba/index.html", "DirectBasicAuthClient", router);
> /rest-jwt/index.html link:
> 
> If I click on the link, I get a "unauthorized" error: OK.
> If I had "?token=" in the url, I get an internal error: OK, this is the expected behaviour, but the error page is not the right one (I expected: error500.html).
> If I provide a good JWT, it works: the profile is properly displayed on the specific page. Though, when I come back to the index page, the profile is still displayed meaning it has been saved in the session while it should have only been available in the request (by default) for this kind of call.
> —
> Reply to this email directly or view it on GitHub <https://github.com/pac4j/vertx-pac4j-demo/pull/35/files#r48539623>.
>